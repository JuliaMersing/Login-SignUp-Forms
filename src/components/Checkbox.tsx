import React, { FormEvent } from 'react';

type CheckboxProps = {
  onChange: (e: FormEvent) => any,
  check: boolean,
}

const Checkbox = ({
    onChange,
    check,
}: CheckboxProps) => (
  <div className="flex items-center">
    <input
      id="remember-me"
      name="remember-me"
      type="checkbox"
      className="h-4 w-4 text-indigo-600 focus:ring-indigo-500 border-gray-300 rounded"
      onChange={onChange}
      defaultChecked={check}
    />
    <label htmlFor="remember-me" className="ml-2 block text-sm text-gray-900"> Remember me </label>
  </div>
);

export default Checkbox;
