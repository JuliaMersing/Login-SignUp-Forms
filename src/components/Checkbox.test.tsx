import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Checkbox from './Checkbox';

describe('Checkbox component', () => {
  const setUp = () => {
    const mockHandleRememberChange = jest.fn();

    const utils = render(<Checkbox onChange={mockHandleRememberChange} check />);
    return {
      ...utils,
      mockHandleRememberChange,
    };
  };

  test('should match snapshot', () => {
    setUp();
    expect(screen.getByRole('checkbox')).toMatchSnapshot();
  });

  test('should call rememberChange', async () => {
    const { mockHandleRememberChange, getByRole } = setUp();

    const remember = (await getByRole('checkbox')) as HTMLInputElement;
    await userEvent.click(remember);
    expect(mockHandleRememberChange).toHaveBeenCalledTimes(1);
  });

  test('should change rememberChange when remember checkbox is checked', async () => {
    const { getByRole } = setUp();

    const remember = (await getByRole('checkbox')) as HTMLInputElement;
    await userEvent.click(remember);
    expect(remember.checked).toEqual(false);

    await userEvent.click(remember);
    expect(remember.checked).toEqual(true);
  });
});
