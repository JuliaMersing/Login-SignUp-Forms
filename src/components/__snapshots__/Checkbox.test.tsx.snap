// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component should match snapshot 1`] = `
<input
  checked=""
  class="checkbox-input"
  id="remember-me"
  name="remember-me"
  type="checkbox"
/>
`;
